android/KMAPro/kMAPro/src/main/java/com/keyman/android/CheckInstallReferrer.java

```diff
@@ -59,6 +64,35 @@ public static void checkGooglePlayInstallReferrer(MainActivity mainActivity, Con
     editor.putBoolean(hasGooglePlayInstallReferrerBeenCheckedKey, true);
     editor.commit();
 
+    // local environment or test builds are nearly always side loaded
+    if (BuildConfig.VERSION_ENVIRONMENT.equalsIgnoreCase("local") ||
+        BuildConfig.VERSION_NAME.matches("^.*(-test-\\d+)$")) {
+      return;
+    }
+
+    // Determine if Keyman installed from Google Play store
+    PackageManager packageManager = context.getPackageManager();
+    String installerInfo = null;
+    try {
+      String packageName = context.getPackageName();
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        installerInfo = packageManager.getInstallSourceInfo(packageName).getInstallingPackageName();
+      } else {
+        installerInfo = packageManager.getInstallerPackageName(packageName);
+      }
+    } catch (PackageManager.NameNotFoundException e) {
+      KMLog.LogException(TAG, "Error determining packageManager", e);
+      return;
+    }
+
+    if (installerInfo == null) {
+      // Side loaded so just return
+      return;
+    } else if (!installerInfo.equalsIgnoreCase(GOOGLE_PLAY)) {
```

Comment:
```suggestion
    }
    
    if (!installerInfo.equalsIgnoreCase(GOOGLE_PLAY)) {
```

Given we are `return`ing above there's no need for the `else`.